charts/victoria-metrics-k8s-stack/hack/sync_grafana_dashboards.py

```diff
@@ -86,246 +63,148 @@ def new_representer(dumper, data):
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  namespace: {{ .Release.Namespace }}
-  name: {{ printf "%%s-%%s" (include "victoria-metrics-k8s-stack.fullname" $) "%(name)s" | trunc 63 | trimSuffix "-" | trimSuffix "." }}
-  labels:
-    {{- if $.Values.grafana.sidecar.dashboards.label }}
-    {{ $.Values.grafana.sidecar.dashboards.label }}: "1"
-    {{- end }}
-    {{- if $.Values.grafana.sidecar.dashboards.additionalDashboardLabels }}
-    {{- range $key, $val := .Values.grafana.sidecar.dashboards.additionalDashboardLabels }}
-    {{ $key }}: {{ $val | quote }}
-    {{- end }}
-    {{- end }}
-    app: {{ include "victoria-metrics-k8s-stack.name" $ }}-grafana
-    {{- include "victoria-metrics-k8s-stack.labels" $ | nindent 4 }}
-    {{- if $.Values.grafana.sidecar.dashboards.additionalDashboardAnnotations }}
-  annotations:
-    {{- range $key, $val := .Values.grafana.sidecar.dashboards.additionalDashboardAnnotations }}
-    {{ $key }}: {{ $val | quote }}
-    {{- end }}
-    {{- end }}
-data:
-{{- end }}
-'''
-
-
-def init_yaml_styles():
-    represent_literal_str = change_style('|', SafeRepresenter.represent_str)
-    yaml.add_representer(LiteralStr, represent_literal_str)
-
-
 def escape(s):
-    return s.replace("{{", "{{`{{").replace("}}", "}}`}}").replace("{{`{{", "{{`{{`}}").replace("}}`}}", "{{`}}`}}")
-
-
-def unescape(s):
-    return s.replace("\{\{", "{{").replace("\}\}", "}}")
-
-
-def yaml_str_repr(struct, indent=2):
-    """represent yaml as a string"""
-    text = yaml.dump(
-        struct,
-        width=1000,  # to disable line wrapping
-        default_flow_style=False  # to disable multiple items on single line
+    return (s
+        .replace('{{', '{{`{{')
+        .replace('}}', '}}`}}')
+        .replace('{{`{{', '{{`{{`}}')
+        .replace('}}`}}', '{{`}}`}}')
+        .replace('[[', '{{')
+        .replace(']]', '}}')
     )
-    text = escape(text)  # escape {{ and }} for helm
-    text = unescape(text)  # unescape \{\{ and \}\} for templating
-    text = textwrap.indent(text, ' ' * indent)
-    return text
 
+def init_yaml_styles():
+    represent_literal_str = change_style('|', SafeRepresenter.represent_str)
+    quoted_presenter = lambda dumper, data : dumper.represent_scalar('tag:yaml.org,2002:str', data, style="'")
+    yaml.add_representer(literal, represent_literal_str)
+    yaml.add_representer(quoted, quoted_presenter)
+
+def fix_expr(target):
+    """Remove trailing whitespaces and line breaks, which happen to creep in
+     due to yaml import specifics;
+     convert multiline expressions to literal style, |-"""
+    if 'expr' in target:
+        target['expr'] = target['expr'].rstrip()
+        if '\n' in target['expr']:
+            target['expr'] = literal(target['expr'])
 
 def replace_ds_type_in_panel(panel):
+    if 'gridPos' in panel:
+        if 'y' in panel['gridPos']:
+            y = panel['gridPos']['y']
+            del panel['gridPos']['y']
+            panel['gridPos']['[[ .yaxis ]]'] = y
```

Comment:
Could you help adding some comments here, to help understanding the issue for future reader?

Reply:
done